cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -147,15 +124,13 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
                            String.format("Namespace '%s' created successfully.", namespaceId));
     } catch (AlreadyExistsException e) {
       responder.sendString(HttpResponseStatus.OK, String.format("Namespace '%s' already exists.", namespaceId));
-    } catch (Exception e) {
-      LOG.error("Internal error while creating namespace.", e);
-      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage());
     }
   }
 
   @DELETE
   @Path("/unrecoverable/namespaces/{namespace-id}")
-  public void delete(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespace) {
+  public void delete(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespace)
+    throws Exception {
     if (!cConf.getBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, Constants.Dangerous.DEFAULT_UNRECOVERABLE_RESET)) {
       responder.sendString(HttpResponseStatus.FORBIDDEN,
```

Comment:
not this PR, but maybe ``HttpExceptionHandler`` should handle ``FORBIDDEN`` as well?